packages/core/core/src/TargetResolver.js

```diff
@@ -176,7 +176,16 @@ export default class TargetResolver {
             // For serve, write the `dist` to inside the parcel cache, which is
             // temporary, likely in a .gitignore or similar, but still readily
             // available for introspection by the user if necessary.
-            distDir: path.resolve(this.options.cacheDir, DEFAULT_DIST_DIRNAME),
+            // unless there are no package targets in which case we write to the ./dist consistent with watch
+            // @see https://github.com/parcel-bundler/parcel/issues/4518
+            distDir:
+              this.options.distDir ??
+              path.resolve(
```

Comment:
This should never happen now, because you defaulted `this.options.distDir` in resolveOptions.js.

Reply:
That was left from before I defaulted options.distDir. I removed this code.

ParcelOptions.distDir is now a required value so downstream consumers can rely on it and not choose or hardcode their own directories.